### ISSUE TITLE
Write the model training loop

### ISSUE THREAD

**Issue by joyyy-99** [joyyy-99/GRP-A-ISP-joyyy-99#7]
Define the loss function and the optimizer.
Implement the training loop with validation steps after each epoch.
